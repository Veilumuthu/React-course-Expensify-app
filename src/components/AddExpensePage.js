import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses'

const AddExpensePage = ({startAddExpense, history}) => {
    const onSubmit = (expense) => {
        startAddExpense(expense)
        history.push('/');
    }
    return (
        <div>
            <div className = "page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Add Expense</h1>
                </div>
            </div>
            <div className="content-container">
                <ExpenseForm onSubmit={onSubmit} />
            </div>
        </div>
    )
}
// export class AddExpensePage extends React.Component {

//     render() {
//         return (

//         )
//     }
// }

const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
})
export default connect(undefined, mapDispatchToProps)(AddExpensePage);