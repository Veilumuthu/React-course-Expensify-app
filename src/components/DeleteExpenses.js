import React from 'react';
import {startDeleteExpense} from '../actions/expenses';
import {connect} from 'react-redux';

class DeleteExpenses extends React.Component {
    onDelete = (expense) => {
        this.props.startDeleteExpense(expense)
    }
    render () {
        return (
            <div className="content-container">
                <button className="button--delete" onClick= {this.onDelete}>Delete Expenses</button>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    startDeleteExpense: (expense) => dispatch(startDeleteExpense(expense))
})
export default connect(undefined, mapDispatchToProps)(DeleteExpenses);