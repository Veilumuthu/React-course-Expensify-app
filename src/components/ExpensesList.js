import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectorExpense from '../selectors/expenses'
// for tesing
export const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        {props.expenses.map((expense) => {
            return <ExpenseListItem  key = {expense.id} {...expense}/>
        })}
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectorExpense(state.expenses, state.filters)
    };
}
export default connect(mapStateToProps)(ExpenseList);
