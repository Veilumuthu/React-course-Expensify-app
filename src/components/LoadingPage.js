import React from 'react';

const Loadingpage = () => (
    <div className="loader">
        <img className="loader__image" src="/images/771.gif" />
    </div>
);

export default Loadingpage;