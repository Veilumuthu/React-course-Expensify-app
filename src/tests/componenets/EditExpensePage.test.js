import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { EditExpensePage } from '../../components/EditExpensePage';
import '../setupTests';

let editExpense, removeExpense, history, wrapper;

beforeEach(() => {
    editExpense = jest.fn();
    removeExpense = jest.fn();
    history = { push: jest.fn()};
    wrapper = shallow(<EditExpensePage 
        editExpense={editExpense}
        removeExpense={removeExpense}
        history={history}
        expenses={expenses[2]} /> );
})

test('sholud render editexpense page', () => {
    expect(wrapper).toMatchSnapshot();
})

// test('Sholud handle editexpense', () => {
//     wrapper.find('ExpenseForm').prop('onSubmit')(expenses[2]);
//     expect(history.push).toHaveBeenLastCalledWith('/');
//     expect(editExpense).toHaveBeenLastCalledWith(expenses[2].id, expenses);
// })

// test('Should handle removeexpense', () => {
//     wrapper.find('button').simulate('click');
//     expect(history.push).toHaveBeenLastCalledWith('/');
//     expect(removeExpense).toHaveBeenLastCalledWith({
//         id: expenses[2].id
//     });
// })