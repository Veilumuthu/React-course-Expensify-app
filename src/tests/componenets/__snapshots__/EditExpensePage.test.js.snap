// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sholud render editexpense page 1`] = `
<div>
  <ExxpenseForm
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
